earthfile2llb/interpreter.go

```diff
@@ -1239,6 +1248,79 @@ func (i *Interpreter) handleBuild(ctx context.Context, cmd spec.Command, async b
 	return nil
 }
 
+func (i *Interpreter) handleWildcardBuilds(ctx context.Context, fullTargetName string, cmd spec.Command, async bool) error {
+
+	parsedTarget, err := domain.ParseTarget(fullTargetName)
+	if err != nil {
+		return i.wrapError(err, cmd.SourceLocation, "failed to parse target")
+	}
+
+	dir, base := filepath.Split(parsedTarget.GetLocalPath())
+	if strings.Contains(dir, "*") || base != "*" {
+		return i.errorf(cmd.SourceLocation, "wildcard BUILD pattern must end with a single '*'")
+	}
+
+	isRemoteTarget := i.target.IsRemote()
+
+	var matches []string
+
+	if isRemoteTarget {
+		matches, err = i.converter.ExpandRemoteWildcard(ctx, i.target, parsedTarget.GetLocalPath())
+	} else {
+		matches, err = fileutil.GlobDirs(parsedTarget.GetLocalPath())
+	}
+	if err != nil {
+		return i.wrapError(err, cmd.SourceLocation, "invalid BUILD wildcard pattern")
+	}
+
+	children := []spec.Command{}
+	for _, match := range matches {
+		childTargetName := fmt.Sprintf("./%s+%s", match, parsedTarget.GetName())
+
+		childTarget, err := domain.ParseTarget(childTargetName)
+		if err != nil {
+			return i.wrapError(err, cmd.SourceLocation, "failed to parse target %q", childTargetName)
+		}
+
+		data, _, _, err := i.converter.ResolveReference(ctx, childTarget)
+		if err != nil {
+			notExist := buildcontext.ErrEarthfileNotExist{}
+			if errors.As(err, &notExist) {
+				continue
```

Comment:
So if an Earthfile does not exist in a directory, it's not an error, right? Can you confirm that this works for both local and remote cases?

Reply:
That's right. The path is ignored in that case. I'll add some more tests. There was one at one point, but I think I dropped it by mistake while refactoring.

Reply:
Oh, the directory was missing a `.gitkeep`. Thanks.